pybaseball/statcast_pitcher.py

```diff
@@ -87,3 +88,19 @@ def statcast_pitcher_percentile_ranks(year: int) -> pd.DataFrame:
     data = pd.read_csv(io.StringIO(res.decode('utf-8')))
     # URL returns a null player with player id 999999, which we want to drop
     return data.loc[data.player_name.notna()].reset_index(drop=True)
+
+@cache.df_cache()
+def statcast_pitcher_spin_dir_comp(year: int, pitch_combo: str = "4 - Seamer / Changeup", minP: int = 100, pov: str = "Pit") -> pd.DataFrame:
+    pitches = ["4-Seamer", "Changeup", "Curveball", "Slider", "Sinker", "Cutter"]
+    pitch_combos = [' / '.join(perm) for perm in permutations(pitches, 2)]
+    if pitch_combo not in pitch_combos:
+        print(f"Not a valid pitch combo. Must include two of {pitches}, separated by ' / '.")
+        return
```

Comment:
As well I'd suggest doing all string comparisons as lower case. No sense in erroring if someone used `curveball` instead of `Curveball`

Reply:
Cool. The returned data has `pitch_a` and `pitch_b`, so I'll use that nomenclature and will do the string comparisons in lowercase. Good call there

Reply:
Here's a concrete suggestion. The idea is that all the pitch codes stuff would just use this normalize call. I know it's a bit longwinded, but it does result in clearer error messages ('ALL' especially can cause some trippy stuff)
```python
pitch_codes = ["FF", "SIFT", "CH", "CUKC", "FC", "SL", "FS", "ALL"] # note: all doesn't work in words, we'll have some special handling
pitch_names = ["4-Seamer", "Sinker", "Changeup", "Curveball", "Cutter", "Slider", "Sinker"]
pitch_names_upper = [p.upper() for p in pitch_names]

# including all the codes to themselves makes this simpler later
name_to_code_map = dict(zip(pitch_codes + pitch_names_upper, pitch_codes + pitch_codes))
code_to_name_map = dict(zip(pitch_codes, pitch_names))

def _norm_pitch_code(pitch: str, to_word: bool = False) -> str:
    # everything goes to code first to keep the code simple, since dict lookups are cheap
    try:
        pitch = name_to_code_map.get(pitch.upper())
    except KeyError:
        raise ValueError(f'{pitch} is not a valid pitch!')
    if to_word:
        if pitch == 'ALL':
            raise ValueError("'All' cannot be used in this function")
        pitch = code_to_name_map[pitch]
    return pitch
```

Reply:
A potential alternative could be:
```python
def _norm_pitch_code(pitch: str, to_word: bool = False) -> str:
    normed = name_to_code_map.get(pitch.upper())
    normed = code_to_name_map.get(normed) if to_word else normed
    if normed is None:
        raise ValueError(f'{pitch} is not a valid pitch!') # this is funky for all though
    return normed
```
This is much shorter, but it leads to weird messaging on the corner case of 'all' for a function that needs names, which don't accept 'all'

Reply:
Another potential alternative could be:
```python
def _norm_pitch_code(pitch: str, to_word: bool = False) -> str:
    normed = name_to_code_map.get(pitch.upper())
    normed = code_to_name_map.get(normed) if to_word else normed
    if normed is None:
        if pitch.lower() is 'all':
            raise ValueError("'All' is not a valid pitch in this particular context!")
        raise ValueError(f'{pitch} is not a valid pitch!') # this is funky for all though
    return normed
```
This covers that edge case

Reply:
I like that last one. More succinct but covers everything. Would we want `_norm_pitch_code()` and the maps to live in `statcast_pitcher` or `utils`?